modules/helm/Makefile.build

```diff
@@ -71,13 +71,14 @@ helm/chart/build/deps:
 	@echo "Fetch dependencies for $(HELM_PACKAGE_PATH)$(CHART_NAME)"
 	@$(HELM) dependency build --debug $(HELM_PACKAGE_PATH)$(CHART_NAME)
 
-# Create a helm package called CHART_NAME and store it in HELM_PACKAGE_PATH
+# Create a helm package called CHART_NAME and store it in HELM_PACKAGE_PATH optionally pinning it with IMAGE_VERSION
 helm/chart/build/package:
 	$(call assert-set,CHART_NAME)
 	$(call assert-set,TARGET_VERSION)
 	$(call assert-set,HELM_PACKAGE_PATH)
-	@echo "For back compatibility pinning image.tag to $(TARGET_VERSION)"
-	@set -o pipefail; yq write --inplace $(HELM_PACKAGE_PATH)$(CHART_NAME)/values.yaml image.tag $(TARGET_VERSION) | head -1
+	$(eval IMAGE_VERSION ?= $(TARGET_VERSION))
```

Comment:
@lhw is it possible to avoid `eval` ?

Reply:
My aim here was to do as few changes as possible to make this change happen. As far as I can tell there is no shorter way to do this in Makefiles. You could of course duplicate the lines and do a conditional run. But from my experience this is far from the worst use-case for eval()

Reply:
What if we deferred to bash instead?
```
set -o pipefail; yq write --inplace $(HELM_PACKAGE_PATH)$(CHART_NAME)/values.yaml image.tag $${IMAGE_VERSION:-TARGET_VERSION} | head -1
```

Reply:
the `:-` operator in the bash variable interpolation will then set the default value if one is not set. 

We like to avoid `eval` at all cost in `Makefiles` because the flow of is not linear.

Reply:
I mean sure. But that does create an external dependency where none is needed. I am not all that invested in this change. I just did a quick fix for a co-worker who wanted the option. If you think that eval() is an issue feel free to change it.

Reply:
Thanks @lhw. Not trying to be difficult =) Just rather than evaluate on a case-by-case basis whether or not `eval` makes sense in some particular use-case, we outright avoid it altogether. It reduces decisions.

https://docs.cloudposse.com/tools/make/best-practices/